### PR TITLE
Added Contact Us Component

### LINKED ISSUE #8: [Component]: Create a Contact Us Form Component
#### Description
Develop a "Contact Us" form component where users can submit their inquiries. The form should include fields for name, email, and message, and provide appropriate error handling for invalid inputs.

**Form Now entries enter into this wont be sent/redirected anywhere, it will be part of seperate issue**

#### Approach
- Create a `ContactUsForm` component.
- Use controlled inputs to manage form state.
- Include fields:
  - **Name** (required, text input)
  - **Email** (required)
  - **Message** (required, textarea with a character limit).
- Implement basic form validation to ensure required fields are filled and inputs are valid.

### Additional Context
- Note create this as a component and not in main Home page file. Later this will be aligned with SocialMediaIcons component (ref issue #7 )
- Refer to the **Figma design** for layout and styling: [Figma Link](https://www.figma.com/design/kHauIvxVeOb8diy8jO6jJd/Website?node-id=0-1&t=VaLTx4vGpSXga6dU-1).

### Relevant Images/ScreenShots
image

### PR DESCRIPTION
**Pull request by PrashamMehta-04**
# Welcome to the PClub WOC 4.0 Project! 👋🎉

Thanks for taking the time to fill out this Pull Request Template.

---

## Pull Request Title
Give an appropriate and relevant title to your Pull Request [PR].

---

## Related Issue
Solves/Resolves/Fixes #8.

---

## Description
Provide a detailed explanation of the changes made in this Pull Request [PR].
- Describe the problem or feature being addressed in it.
- Highlight key modifications in the codebase.

---

## Checklist
Please ensure the following before submitting the PR:
- [ ] I have read the `CODE_OF_CONDUCT.md` file and followed it's guidlines.
- [ ] I have followed the design specified in the Figma file (if applicable).

---

## Additional Comments
Include any additional notes or comments for the reviewers. If applicable, mention dependencies, challenges faced, or potential improvements.

### CODE REVIEW

File: src/components/ContactUsForm/Contact_Us.css [hunk 1]
```diff
@@ -0,0 +1,101 @@
+:root {
+    font-family: Inter, system-ui, Avenir, Helvetica, Arial, sans-serif;
+    line-height: 1.5;
+    font-weight: 400;
+  
+    color-scheme: light dark;
```
Comment (Saumya40-codes) [inline src/components/ContactUsForm/Contact_Us.css]: we won't be needing this `color-scheme` style

File: src/components/ContactUsForm/Contact_Us.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import "./Contact_Us.css";
+
+const ContactUs = () => {
```
Comment (Saumya40-codes) [inline src/components/ContactUsForm/Contact_Us.jsx:4]: this actually doesnt match the figma design, https://www.figma.com/design/kHauIvxVeOb8diy8jO6jJd/Website?node-id=124-5&t=KYWacSRJGK7iiUoH-0

We can go in such manner

```
<label> Full Name </label>
<input />

 <label> Email </label>
<input />
```

**Comment by PrashamMehta-04** [comment]
Updated both the files by making the necessary changes..

**Comment by Saumya40-codes** [comment]
@PrashamMehta-04 the implementation is nearly right, but

the figma design looks like this:
image

and current (your implementation) is somewhat  like this:
image


can you please take a look at some suggested changes(review) below

File: src/components/ContactUsForm/Contact_Us.css [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  a {
+    font-weight: 500;
+    color: #646cff;
+    text-decoration: inherit;
+  }
+  a:hover {
+    color: #535bf2;
+  }
+  
+  body {
```
Comment (Saumya40-codes) [inline src/components/ContactUsForm/Contact_Us.css]: we can remove this body style as there is no need for any explicit body design

File: src/components/ContactUsForm/Contact_Us.css [hunk 4]
```diff
@@ -0,0 +1,97 @@
+:root {
+    font-family: Inter, system-ui, Avenir, Helvetica, Arial, sans-serif;
+    color: rgba(255, 255, 255, 0.87);
+    background-color: #242424;
```
Comment (Saumya40-codes) [inline src/components/ContactUsForm/Contact_Us.css]: We wont be needing that as the background color of this form and the original bg-color of the site is same so no need to state explicitly 

we can remove

```
color: rgba(255, 255, 255, 0.87);
background-color: #242424;
```

File: src/components/ContactUsForm/Contact_Us.css [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  }
+  button:focus,
+  button:focus-visible {
+    outline: 4px auto -webkit-focus-ring-color;
+  }
+  .contact-us-container {
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    background-color: #0a0a23;
```
Comment (Saumya40-codes) [inline src/components/ContactUsForm/Contact_Us.css]: we can remove this propery of `background-color` from here and add some fix width and height like this

```
max-width: 536px;
max-height: 535px;

```

File: src/components/ContactUsForm/Contact_Us.css [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    margin-bottom: 10px;
+  }
+  
+  .contact-form input,
+  .contact-form textarea {
+    padding: 10px;
+    margin-top: 10px;
+    border: 10px;
+    border-radius: 10px;
+    width: 300px;
```
Comment (Saumya40-codes) [inline src/components/ContactUsForm/Contact_Us.css]: we change this to  (as we already imposed max-width to the parent container above)
```
width: 100%
```

additionally please add following style here

```
  padding: 20px;
  outline: none;
  background-color: transparent;
```

and after this style, create this one style

```
.contact-form input, .contact-form textarea {
    border: none;
    border-bottom: 1px solid #ccc;
    border-radius: 0;
  }
```

File: src/components/ContactUsForm/Contact_Us.jsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import "./Contact_Us.css";
+
+const ContactUs = () => {
+  return (
+    <div className="contact-us-container">
+      <div className="contact-form">
+        <h1>Contact Us</h1>
+        <form>
+          <label>Full Name</label>
```
Comment (Saumya40-codes) [inline src/components/ContactUsForm/Contact_Us.jsx]: now, we can keep this as what it was before i.e. 

```
<input placeholder="Full Name" />
 <input placeholder="Email"/>   
```

(no need for label fields with input with current styles :)

**Comment by Saumya40-codes** [comment]
Its perfect 🙌 but just implement the suggestions above to keep it more align with figma design.

**Comment by PrashamMehta-04** [comment]
Thank you for clarifing and helping me out. I have made the mentioned changes...

File: src/components/ContactUsForm/Contact_Us.css [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    background-color: transparent;
+    border: none;
+    border-bottom: 1px solid #ccc;
+    border-radius: 0;
+  }
+  
+  
+  .contact-form button {
+    padding: 10px;
+    background-color: #00bfff;
```
Comment (Saumya40-codes) [inline src/components/ContactUsForm/Contact_Us.css]: this should be #004457   (re: figma design)

File: src/components/ContactUsForm/Contact_Us.css [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    display: flex;
+    flex-direction: column;
+  }
+  
+  .contact-form label {
+    margin-bottom: 10px;
+  }
+  
+  .contact-form input,
+  .contact-form textarea {
```
Comment (Saumya40-codes) [inline src/components/ContactUsForm/Contact_Us.css:69]: currently when we write, by default color: black
image


for visibility, we should change it to white (#fff) by adding the same in this style.

**Comment by Saumya40-codes** [comment]
LGTM overall 🚀
Thanks for keeping up with the changes and contributing.

**Comment by PrashamMehta-04** [comment]
Changes made.
Thank you for the opportunity.

**Pull request merged**